kpops/pipeline_generator/pipeline.py

```diff
@@ -59,6 +64,10 @@ def __iter__(self) -> Iterator[PipelineComponent]:
     def __len__(self) -> int:
         return len(self.components)
 
+    def validate_graph(self) -> None:
+        if not nx.is_directed_acyclic_graph(self.graph):
+            raise ValueError("Pipeline contains cycles.")
```

Comment:
didn't you want to change the error message to say "pipeline is not a valid DAG" or similar? because it might have different issues than only cycles